core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -139,7 +138,23 @@ class RangePolicy : public Impl::PolicyTraits<Properties...> {
     check_conversion_safety(work_end);
     check_bounds_validity();
     set_auto_chunk_size();
-    set(args...);
+  }
+
+  template <typename IndexType1, typename IndexType2,
+            std::enable_if_t<(std::is_convertible_v<IndexType1, member_type> &&
+                              std::is_convertible_v<IndexType2, member_type>),
+                             bool> = false>
+  inline RangePolicy(const typename traits::execution_space& work_space,
+                     const IndexType1 work_begin, const IndexType2 work_end,
+                     const ChunkSize chunk_size)
+      : m_space(work_space),
+        m_begin(work_begin),
+        m_end(work_end),
+        m_granularity(chunk_size.value),
+        m_granularity_mask(m_granularity - 1) {
+    check_conversion_safety(work_begin);
+    check_conversion_safety(work_end);
+    check_bounds_validity();
```

Comment:
```suggestion
    check_bounds_validity();
    set_chunk_size(chunk_size.value);
```

Reply:
I'm setting the chunk size in the member initializers for `m_granularity` and `m_granularity_mask`.  I want to confirm with you that you'd rather I do it via the `set_chunk_size(int)` function.

Reply:
I would since technically this avoid repeating the logic.

Reply:
Done.  I also removed setting it in the member initialisers (it would be misleading to set it in both places).

Reply:
>  I also removed setting it in the member initialisers (it would be misleading to set it in both places).

I think you forgot to push these changes.  But whatever, this can be done as a follow up.